src/main/java/net/sf/jabref/logic/importer/fetcher/IsbnFetcher.java

```diff
@@ -72,20 +68,23 @@ public HelpFile getHelpPage() {
                 //Downloads the source code of the site and then creates a .bib file out of the String
                 URLDownload urlDownload = new URLDownload(url);
                 String bibtexString = urlDownload.downloadToString(StandardCharsets.UTF_8);
-                BibEntry entry = BibtexParser.singleFromString(bibtexString, ImportFormatPreferences.fromPreferences(Globals.prefs));
-
-                entry.clearField(FieldName.URL);
+                Optional<BibEntry> entry = BibtexParser.singleFromStringOptional(bibtexString, ImportFormatPreferences.fromPreferences(Globals.prefs));
 
-                //Removes every non-digit character in the PAGETOTAL field.
-                if (entry.hasField(FieldName.PAGETOTAL)) {
-                    String pagetotal = entry.getFieldOptional(FieldName.PAGETOTAL).get();
-                    entry.setField(FieldName.PAGETOTAL, pagetotal.replaceAll("[\\D]", ""));
-                }
+                if (entry.isPresent()) {
+                    BibEntry bibEntry = entry.get();
+                    if (bibEntry.hasField(FieldName.URL)) {
+                        bibEntry.clearField(FieldName.URL);
+                    }
 
-                result = Optional.ofNullable(entry);
+                    //Removes every non-digit character in the PAGETOTAL field.
+                    Optional<String> pagetotal = bibEntry.getFieldOptional(FieldName.PAGETOTAL);
+                    if (pagetotal.isPresent()) {
```

Comment:
Just a small thing, but here the pagetotal.ifPresent(Consumer) would make sense as you only have on single statement.